janus_core/helpers/mlip_calculators.py

```diff
@@ -6,17 +6,72 @@
 - https://github.com/Quantum-Accelerators/quacc.git
 """
 
-from typing import get_args
+from pathlib import Path
+from typing import Any, Optional, Union, get_args
 
 from ase.calculators.calculator import Calculator
 import torch
 
-from janus_core.helpers.janus_types import Architectures, Devices
+from janus_core.helpers.janus_types import Architectures, Devices, PathLike
+
+
+def _set_model_path(
+    model_path: Optional[PathLike] = None,
+    kwargs: Optional[dict[str, Any]] = None,
+) -> Optional[Union[PathLike, torch.nn.Module]]:
+    """
+    Set `model_path`.
+
+    Parameters
+    ----------
+    model_path : Optional[PathLike]
+        Path to MLIP file.
+    kwargs : Optional[dict[str, Any]]
+        Dictionary of additional keyword arguments passed to the selected calculator.
+
+    Returns
+    -------
+    Optional[Union[PathLike, torch.nn.Module]]
+        Path to MLIP model file, loaded model, or None.
+    """
+    kwargs = kwargs if kwargs else {}
+
+    # kwargs that may be used for `model_path`` for different MLIPs
+    # Note: "model" for chgnet (but not mace_mp or mace_off) and "potential" may refer
+    # to loaded PyTorch models
+    model_path_kwargs = ("model", "model_paths", "potential", "path")
+
+    # Use model_path if specified, but check not also specified via kwargs
+    if model_path:
+        if any(kwarg in kwargs for kwarg in model_path_kwargs):
+            raise ValueError(
+                "`model_path` cannot be used in combination with 'model', "
+                "'model_paths', 'potential', or 'path'"
+            )
+    else:
+        # Check at most one suitable kwarg is specified
+        if sum(kwarg in kwargs for kwarg in model_path_kwargs) > 1:
+            raise ValueError(
+                "Only one of 'model', 'model_paths', 'potential', and 'path' can be "
+                "specified"
+            )
+
+        # Set model_path from kwargs if any are specified
+        for kwarg in model_path_kwargs:
+            if kwarg in kwargs:
+                model_path = kwargs.pop(kwarg, None)
+
+    # Convert to path if file/directory exists
+    if isinstance(model_path, (Path, str)) and Path(model_path).expanduser().exists():
```

Comment:
Why do you need to `expanduser` here?

Reply:
So that the path can be wrapped in quotes e.g. in the CLI `--model-path "~/.cache/mace/5f5yavf3"` didn't work, but `--model-path ~/.cache/mace/5f5yavf3` did, which didn't seem very intuitive, and may be annoying with special characters

Reply:
I meant in this line in particular, does `Path` not resolve that correctly to check for existence. If not, that's interesting.

Reply:
```
In [22]: Path("~/.cache/mace/5f5yavf3").expanduser().exists()
Out[22]: True

In [23]: Path("~/.cache/mace/5f5yavf3").exists()
Out[23]: False
```
Is this what you mean?

Reply:
Yep. Weird, but I think that settles this.